test/metadata-retrieval/metadataretrieval.go

```diff
@@ -0,0 +1,337 @@
+package metadataretrieval
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"text/tabwriter"
+	"time"
+
+	"github.com/src-d/regression-retrieval/prometheus"
+	"github.com/src-d/regression-retrieval/test"
+
+	"github.com/src-d/regression-core"
+	"gopkg.in/src-d/go-log.v1"
+	"gopkg.in/yaml.v2"
+)
+
+type (
+	metadataRetrievalResults map[string][]*Result
+	versionResults           map[string]metadataRetrievalResults
+
+	// Test holds the information about a metadata-retrieval test
+	Test struct {
+		config            regression.Config
+		metadataRetrieval map[string]*regression.Binary
+		// organizations is array of lists of coma-separated organizations
+		organizations []string
+		results       versionResults
+		log           log.Logger
+	}
+)
+
+// Kind is an identifier of util type to be tested, used in factory test constructor
+const Kind = "metadata-retrieval"
+
+func init() {
+	test.Register(Kind, NewTest)
+}
+
+// Result is a wrapper around regression.Result that additionally contains organizations that were processed
+type Result struct {
+	*regression.Result
+	Organizations string
+}
+
+// NewTest creates a new Test struct
+func NewTest(config regression.Config) (test.Test, error) {
+	l, err := (&log.LoggerFactory{Level: log.InfoLevel}).New(log.Fields{})
+	if err != nil {
+		return nil, err
+	}
+
+	return &Test{
+		config: config,
+		log:    l,
+	}, nil
+}
+
+// Prepare downloads and builds required metadata-retrieval versions
+func (t *Test) Prepare() error {
+	return t.prepareMetadataRetrieval()
+}
+
+func (t *Test) prepareMetadataRetrieval() error {
+	t.log.Infof("Preparing metadata-retrieval binaries")
+	releases := regression.NewReleases("src-d", "metadata-retrieval", t.config.GitHubToken)
+
+	t.metadataRetrieval = make(map[string]*regression.Binary, len(t.config.Versions))
+	for _, version := range t.config.Versions {
+		b := NewMetadataRetrieval(t.config, version, releases)
+		err := b.Download()
+		if err != nil {
+			return err
+		}
+
+		t.metadataRetrieval[version] = b
+	}
+
+	return nil
+}
+
+// RunLoad executes the tests
+func (t *Test) RunLoad() error {
+	results := make(versionResults)
+
+	for _, version := range t.config.Versions {
+		_, ok := results[version]
+		if !ok {
+			results[version] = make(metadataRetrievalResults)
+		}
+
+		metadataRetrieval, ok := t.metadataRetrieval[version]
+		if !ok {
+			panic("metadataRetrieval not initialized. Was Prepare called?")
+		}
+
+		l := t.log.New(log.Fields{"version": version})
+
+		l.Infof("Running version tests")
+
+		times := t.config.Repeat
+		if times < 1 {
+			times = 1
+		}
+
+		// TODO(kyrcha): add a regression.yml file to metadata-retrieval
```

Comment:
let's postpone merging this PR until `regression.yml` appears in `metadata-retrieval`
it is a must

Reply:
Created a [PR](https://github.com/src-d/metadata-retrieval/pull/90)